src/main/cpp/subsystems/Arm.cpp

```diff
@@ -1,41 +1,123 @@
 #include "subsystems/Arm.h"
+#include "Parameters.h"
+#include "ui/UserInterface.h"
 
+//pulls PID values from the parameters.toml file
 bool Arm::Init() {
-    arm_pid_ = std::make_shared<frc::PIDController>(0, 0, 0);
+    
+    double arm_kP = RobotParams::GetParam("arm.lift_pid.kP", 0.005);
+    double arm_kI = RobotParams::GetParam("arm.lift_pid.kI", 0.0);
+    double arm_kD = RobotParams::GetParam("arm.lift_pid.kD", 0.0);
 
-    inches_pid_ = std::make_shared<frc::PIDController>(0, 0, 0);
+    double extend_kP = RobotParams::GetParam("arm.extend_pid.kP", 0.005);
+    double extend_kI = RobotParams::GetParam("arm.extend_pid.kI", 0.0);
+    double extend_kD = RobotParams::GetParam("arm.extend_pid.kD", 0.0);
+
+    arm_pid_ = std::make_shared<frc::PIDController>(arm_kP, arm_kI, arm_kD);
+
+    inches_pid_ = std::make_shared<frc::PIDController>(extend_kP, extend_kI, extend_kD);
 
     arm_lift_output_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/lift_output");
     arm_lift_enc_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/lift_enc");
+    arm_lift_setpoint_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/lift_setpoint");
+
+    arm_extend_output_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/extend_output");
+    arm_extend_enc_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/extend_enc");
+    arm_extend_setpoint_log_ = wpi::log::DoubleLogEntry(TeamOKC::log, "/arm/extend_setpoint");
+
+    this->arm_pid_->SetSetpoint(0);
+    this->inches_pid_->SetSetpoint(0);
+
+    lift_limit = RobotParams::GetParam("arm.lift_limit", 100);
+    extend_limit = RobotParams::GetParam("arm.extend_limit", 100);
```

Comment:
I guess these are private variables without underscores - it's easier to read them as private variables with the trailing underscore